src/mongoose_hooks.erl

```diff
@@ -17,3 +67,98 @@ privacy_check_packet(LServer, Acc, User, PrivacyList, FromToNameType, Dir) ->
+    LServer :: jid:lserver(),
+    Acc :: mongoose_acc:t(),
+    User :: jid:user(),
+    Server :: jid:server(),
+    From :: jid:jid(),
+    Type :: any(),
+    Reason :: any(),
+    Result :: mongoose_acc:t().
+roster_in_subscription(LServer, Acc, User, Server, From, Type, Reason) ->
+    ejabberd_hooks:run_fold(
+        roster_in_subscription,
+        LServer,
+        Acc,
+        [User, Server, From, Type, Reason]).
+
+-spec set_presence_hook(LServer, Acc, LUser, LResource, Presence) -> Result when
+    LServer :: jid:lserver(), Acc :: mongoose_acc:t(), LUser :: jid:luser(),
+    LResource :: jid:lresource(),
+    Presence :: any(),
+    Result :: mongoose_acc:t().
+set_presence_hook(LServer, Acc, LUser, LResource, Presence) ->
+    ejabberd_hooks:run_fold(set_presence_hook, LServer, Acc, [LUser, LServer, LResource, Presence]).
+
+-spec sm_broadcast(LServer, Acc, From, To, Broadcast, SessionCount) -> Result when
+    LServer :: jid:lserver(),
+    Acc :: mongoose_acc:t(),
+    From :: jid:jid(),
+    To :: jid:jid(),
+    Broadcast :: ejabberd_c2s:broadcast(),
+    SessionCount :: non_neg_integer(),
+    Result :: mongoose_acc:t().
+sm_broadcast(LServer, Acc, From, To, Broadcast, SessionCount) ->
+    ejabberd_hooks:run_fold(sm_broadcast, LServer, Acc,
+                            [From, To, Broadcast, SessionCount]).
+
+-spec sm_filter_offline_message(LServer, Drop, From, To, Packet) -> Result when
+    LServer :: jid:lserver(),
+    Drop :: boolean(),
+    From :: jid:jid(),
+    To :: jid:jid(),
+    Packet :: exml:element(),
+    Result :: boolean().
+sm_filter_offline_message(LServer, Drop, From, To, Packet) ->
+    ejabberd_hooks:run_fold(sm_filter_offline_message, LServer,
+                            Drop, [From, To, Packet]).
+
+-spec sm_register_connection_hook(LServer, SID, JID, Info) -> Result when
+    LServer :: jid:lserver(),
+    SID :: 'undefined' | ejabberd_sm:sid(),
+    JID :: jid:jid(),
+    Info :: 'undefined' | [any()],
+    Result :: ok.
+    sm_register_connection_hook(LServer, SID, JID, Info) ->
+    ejabberd_hooks:run_fold(sm_register_connection_hook, LServer, ok,
+                       [SID, JID, Info]).
+
+-spec sm_remove_connection_hook(LServer, Acc, SID, JID, Info, Reason) -> Result when
+    LServer :: jid:lserver(),
+    Acc :: mongoose_acc:t(),
+    SID :: 'undefined' | ejabberd_sm:sid(),
+    JID :: jid:jid(),
+    Info :: ejabberd_sm:info(),
+    Reason :: ejabberd_sm:close_reason(),
+    Result :: mongoose_acc:t().
+sm_remove_connection_hook(LServer, Acc, SID, JID, Info, Reason) ->
+    ejabberd_hooks:run_fold(sm_remove_connection_hook, LServer, Acc,
+                            [SID, JID, Info, Reason]).
+-spec unset_presence_hook(LServer, Acc, LUser, LResource, Status) -> Result when
+    LServer :: jid:lserver(),
+    Acc :: mongoose_acc:t(),
+    LUser :: jid:luser(),
+    LResource :: jid:lresource(),
+    Status :: any(),
```

Comment:
As far as I know the status is a binary representing the status a user put in his presence stanza.